Automattic/wp-calypso
# Publicize Gutenblock: Remove Store

#### Changes proposed in this Pull Request

Everything we're currently doing in the Publicize extension's `store/` directory accesses the `/wp/v2/posts/{ postId }/?context=edit` endpoint. But that's just the posts endpoint, which isn't really specific to Publicize in any way -- all its information can be obtained through posts related selectors.

Thus, this PR removes that store and uses those selectors instead.

#### Testing instructions

* Check out this branch.
* Spin up local Jetpack with the `add/publicize-rest-api-3` branch.
* Make sure Jetpack is active and connected.
* Run `npm run sdk -- gutenberg client/gutenberg/extensions/presets/jetpack/ -- --output-dir=DIR_TO_JETPACK/jetpack/_inc/blocks` where `DIR_TO_JETPACK` is your Jetpack directory.
* Start writing a new post.
* In Gutenberg, click on 'Publish...'
* Open the browser console's network tab, and clear it
* Click on 'Connect new service', which will open a new tab with Publicize connections.
* Close that new tab, landing back in the tab with your new post. In the browser console's network tab, verify that after closing the new tab, the `wp/v2/posts/{postId}/?context=edit` endpoint was hit.
* Verify that the rest of the Publicize extension also still works as before.

Note that there is one weird error that is triggered by `refreshPost()`:

![image](https://user-images.githubusercontent.com/96308/48502457-e0637c00-e83f-11e8-9c96-e158e8b94f9a.png)

which is an upstream bug. I've filed a fix: https://github.com/WordPress/gutenberg/pull/11879
